### ISSUE TITLE
Upgrade build dependencies for GHC 9.4.2

### ISSUE THREAD

**Issue by z0rr0** [bravit/unescaping-print#3]
Is it possible to increase limit for `base` up to 4.16?

```sh
cabal install --lib unescaping-print

Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: unescaping-print-0.1 (user goal)
[__1] next goal: base (dependency of unescaping-print)
[__1] rejecting: base-4.16.3.0/installed-4.16.3.0 (conflict: unescaping-print
=> base>=4.10 && <4.15)
[__1] skipping: base-4.17.0.0, base-4.16.3.0, base-4.16.2.0, base-4.16.1.0,
base-4.16.0.0, base-4.15.1.0, base-4.15.0.0 (has the same characteristics that
caused the previous version to fail: excluded by constraint '>=4.10 && <4.15'
from 'unescaping-print')
[__1] rejecting: base-4.14.3.0, base-4.14.2.0, base-4.14.1.0, base-4.14.0.0,
base-4.13.0.0, base-4.12.0.0, base-4.11.1.0, base-4.11.0.0, base-4.10.1.0,
base-4.10.0.0, base-4.9.1.0, base-4.9.0.0, base-4.8.2.0, base-4.8.1.0,
base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0, base-4.6.0.1,
base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0, base-4.4.0.0,
base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1, base-4.2.0.0,
base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1 (constraint from
non-upgradeable package requires installed instance)
[__1] fail (backjumping, conflict set: base, unescaping-print)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, unescaping-print
```

**Comment by vovs03** [comment]
# Thanks for `unescaping-print`

@bravit thanks for your package: :a: solving print problem with cyrillic-chars.

I want to use it, but get this warning:

```terminal
➜  cabal install --lib unescaping-print
Resolving dependencies...
Error: cabal: Could not resolve dependencies:
[__0] trying: unescaping-print-0.1 (user goal)
[__1] next goal: base (dependency of unescaping-print)
[__1] rejecting: base-4.16.3.0/installed-4.16.3.0 (conflict: unescaping-print
=> base>=4.10 && <4.15)
[__1] skipping: base-4.17.0.0, base-4.16.4.0, base-4.16.3.0, base-4.16.2.0,
base-4.16.1.0, base-4.16.0.0, base-4.15.1.0, base-4.15.0.0 (has the same
characteristics that caused the previous version to fail: excluded by
constraint '>=4.10 && <4.15' from 'unescaping-print')
[__1] rejecting: base-4.14.3.0, base-4.14.2.0, base-4.14.1.0, base-4.14.0.0,
base-4.13.0.0, base-4.12.0.0, base-4.11.1.0, base-4.11.0.0, base-4.10.1.0,
base-4.10.0.0, base-4.9.1.0, base-4.9.0.0, base-4.8.2.0, base-4.8.1.0,
base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0, base-4.6.0.1,
base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0, base-4.4.0.0,
base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1, base-4.2.0.0,
base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1 (constraint from
non-upgradeable package requires installed instance)
[__1] fail (backjumping, conflict set: base, unescaping-print)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, unescaping-print

➜  cabal --version
cabal-install version 3.8.1.0
compiled using version 3.8.1.0 of the Cabal library

➜  ghc --version
The Glorious Glasgow Haskell Compilation System, version 9.2.4
```
Could you fix :b: `unescaping-print` for `base-4.17.0.0`?

**Comment by bravit** [comment]
Hi! I've made the revision on Hackage. Extremely sorry for the delay.
